### PR TITLE
[EFR32] Fix 2 door lock edge cases

### PR DESCRIPTION
**Pull request by mykrupp**
#### Problem
1. When an invalid pinCode is sent with a lock command, and DUT is already locked, a SUCCESS is returned. Expected outcome is that failure is returned because the pinCode is invalid.

2. When clear fabric from credentials is called with the ProgrammingPin credential, an error is returned. Expected outcome is that success is returned.

#### Change overview
1. Remove any checking of lockState in the setLockState function, because the SUCCESS/FAILURE of the function does not depend on the lockState. Also removed a `return true` statement in the case where the current state is the same as the new state command received.

2. Removed the check for the lower bound of the credentialIndex. This is encapsulated in IsValidCredentialIndex, because credentialIndex is unsigned. All door-lock indices are one-indexed, except for ProgrammingPin.. I wonder if this is overlooked in the spec.

#### Testing
Ran through DRLK  TC 2.2 on EFR32, thread network.

### CODE REVIEW

**Review by selissia — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]
Not a fan of the name `IsCredentialIndexGreaterThanZero` since it is not what it return for kProgrammingPIN. 
IsCredentialIndexAllowed or something in that sense would be more adequate to the method behaviour

File: examples/lock-app/efr32/src/LockManager.cpp [hunk 1]
```diff
@@ -104,6 +104,17 @@ bool LockManager::IsValidUserIndex(uint16_t userIndex)
     return (userIndex < kMaxUsers);
 }
 
+bool LockManager::IsCredentialIndexGreaterThanZero(uint16_t credentialIndex, DlCredentialType type)
```
Comment (andy31415) [inline examples/lock-app/efr32/src/LockManager.cpp:107]: The name seems off. it seems to say 'is the credential index greater than zero' which to me translates to 'credentialIndex > 0'.

Please name this something better ... maybe "IsValidCredentialIndex" or something like that.

Comment (mykrupp) [reply examples/lock-app/efr32/src/LockManager.cpp:107]: Yeah, but there's already a function called IsValidCredentialIndex for the upper bound :(

I could change it to isValidCredentialIndexLowerBound, and isValidCredentialIndexUpperBound?

**Comment by mykrupp** [comment]
I ended up just removed the  index > 0 check. This should be taken care in IsValidCredentialIndex, because the indices are unsigned.

The index > 0 check should be encapsulated in `credentialIndex < kMaxCredentialsPerUser`

File: examples/lock-app/efr32/src/LockManager.cpp [hunk 2]
```diff
@@ -406,7 +417,8 @@ bool LockManager::GetCredential(chip::EndpointId endpointId, uint16_t credential
                                 EmberAfPluginDoorLockCredentialInfo & credential)
 {
 
-    VerifyOrReturnValue(credentialIndex > 0, false); // indices are one-indexed
+    VerifyOrReturnValue(IsCredentialIndexGreaterThanZero(credentialIndex, credentialType),
+                        false); // indices are one-indexed except ProgrammingPin
```
[the remark this answers is not in the document]
Comment (mykrupp) [reply examples/lock-app/efr32/src/LockManager.cpp]: removed comment from (index > 0) check

File: examples/lock-app/efr32/include/LockManager.h [hunk 3]
```diff
@@ -158,6 +158,7 @@ class LockManager
                                 uint32_t localEndTime, DlOperatingMode operatingMode);
 
     bool IsValidUserIndex(uint16_t userIndex);
+    bool IsCredentialIndexGreaterThanZero(uint16_t credentialIndex, DlCredentialType type);
```
Comment (jmartinez-silabs) [inline examples/lock-app/efr32/include/LockManager.h:161]: His this prototype still needed then? Seems like you removed it in the source

Comment (mykrupp) [reply examples/lock-app/efr32/include/LockManager.h]: removed the unused function delclaration

**Review by woody-apple — APPROVED** [review]

**Pull request merged**